cmake/FindPackageOrElseAddSubdirectory.cmake

```
@@ -0,0 +1,18 @@
+# Macro for finding a system provided library
+# that fulfills the required minimum version
+# or else including the vendored library
+# from a local git subdirectory
+macro(find_package_or_else_add_subdirectory PACKAGE_NAME MIN_REQUIRED_VERSION SUBDIRECTORY)
+  find_package(${PACKAGE_NAME} ${MIN_REQUIRED_VERSION} QUIET)
+  if (NOT ${${PACKAGE_NAME}_FOUND} OR ("${${PACKAGE_NAME}_VERSION}" VERSION_LESS "${MIN_REQUIRED_VERSION}"))
```

Why is it better than
```suggestion
  if (${${PACKAGE_NAME}_FOUND} AND ("${${PACKAGE_NAME}_VERSION}" VERSION_GREATER_EQUAL "${MIN_REQUIRED_VERSION}"))
```
?

---

Because the current logical expression, expresses the relevant case that we have to handle.
The suggested conditional expression, does not express a relevant case, that is a rather irrelevant case, the fact that we even do something like logging in that scenario is just out of curtesy. We could skip that branch and the code would still work.